types/src/trust_level.rs

```diff
@@ -0,0 +1,40 @@
+use crate::{verification_error, VerificationError};
+
+pub const DEFAULT_TRUST_LEVEL: TrustLevelRatio = TrustLevelRatio::new(1, 3);
+
+pub struct TrustLevelRatio {
+    numerator: u64,
+    denominator: u64,
+}
+
+impl TrustLevelRatio {
+    pub const fn new(numerator: u64, denominator: u64) -> Self {
+        TrustLevelRatio {
+            numerator,
+            denominator,
+        }
+    }
+
+    pub fn numerator(&self) -> u64 {
+        self.numerator
+    }
+
+    pub fn denominator(&self) -> u64 {
+        self.denominator
+    }
+
+    pub fn total_power_needed(
```

Comment:
```suggestion
    pub fn voting_power_needed(
```
I'd name it like so, as there are `total_voting_power` and `voting_power_needed`